lib/hanami/utils/kernel.rb

```diff
@@ -404,15 +413,14 @@ def self.Integer(arg)
       #   # Missing #respond_to?
       #   input = BasicObject.new
       #   Hanami::Utils::Kernel.BigDecimal(input) # => TypeError
-      def self.BigDecimal(arg)
+      def self.BigDecimal(arg, precision: BIGDECIMAL_DEFAULT_PRECISION)
         case arg
+        when ->(a) { a.respond_to?(:to_d) && a.class == Rational } then arg.to_d(precision)
```

Comment:
@AlfonsoUceda I discovered that on MRI `Rational` doesn't respond to `#to_d` and this double check can be perf consuming. What if we flip the case conditionals like this?

```ruby
def self.BigDecimal(arg)
        case arg
        when Float, Complex, Rational
          BigDecimal(arg.to_s)
        when ->(a) { a.respond_to?(:to_d) }
           arg.to_d
        else
          BigDecimal.new(arg)
        end
      rescue NoMethodError
        raise TypeError.new "can't convert #{inspect_type_error(arg)} into BigDecimal"
      end
end
```

This should make the test to pass, but avoids the need for us to decide the precision with that constant. What do you think?